gdal/alg/viewshed.cpp

```diff
@@ -284,25 +276,26 @@ CPLErr GDALViewshedGenerate(GDALRasterBandH hBand, const char* pszTargetRasterNa
                     "Several drivers matching %s extension. Using %s",
                     CPLGetExtension(pszTargetRasterName), aoDrivers[0].c_str());
             }
-            GDALDriverH hDriver = GDALGetDriverByName(aoDrivers[0].c_str());
-            if (hDriver == nullptr)
+
+            GDALDriverManager *hMgr = GetGDALDriverManager();
+            GDALDriver *hDriver = hMgr->GetDriverByName(aoDrivers[0].c_str());
+            if (!hDriver)
             {
                 CPLError(CE_Failure, CPLE_AppDefined,
                     "Cannot get driver for %s", aoDrivers[0].c_str());
                 return CE_Failure;
             }
             /* create output raster */
-            hDstDS = GDALCreate(hDriver, pszTargetRasterName, nXSize, nYStop - nYStart, 1,
-                    GDT_Byte, nullptr);
-            if (hDstDS == nullptr)
+//             GDALDatasetH hDstDS = nullptr;
+            hDstDS = std::unique_ptr<GDALDataset>(hDriver->Create(pszTargetRasterName, nXSize, nYStop - nYStart, 1, GDT_Byte, nullptr));
+            if (!hDstDS)
             {
                 CPLError(CE_Failure, CPLE_AppDefined,
                     "Cannot create dataset for %s", pszTargetRasterName);
                 return CE_Failure;
             }
-
             /* copy srs */
-            GDALSetSpatialRef(hDstDS, GDALGetSpatialRef(hSrcDS));
+            GDALSetSpatialRef(hDstDS.get(), GDALGetSpatialRef(hSrcDS));
```

Comment:
you should rather use the C++ methods each time you use hDstDS. In debug mode, the C handle and C++ pointers  cannot be used without explicit conversion